# BigData Project
# Dataset link : https://www.kaggle.com/jessemostipak/hotel-booking-demand


importing pandas and numpy


In [ ]:

import pandas as pd
import numpy as np

In [9]:
df = pd.read_csv('C:/Users/arsla/OneDrive/Desktop/hotel_bookings.csv',header=0)

In [10]:
df.head(10)

hotel  is_canceled  lead_time  arrival_date_year arrival_date_month  \
0  Resort Hotel            0        342               2015               July   
1  Resort Hotel            0        737               2015               July   
2  Resort Hotel            0          7               2015               July   
3  Resort Hotel            0         13               2015               July   
4  Resort Hotel            0         14               2015               July   
5  Resort Hotel            0         14               2015               July   
6  Resort Hotel            0          0               2015               July   
7  Resort Hotel            0          9               2015               July   
8  Resort Hotel            1         85               2015               July   
9  Resort Hotel            1         75               2015               July   

   arrival_date_week_number  arrival_date_day_of_month  \
0                        27                          1   
1                        27                          1   
2                        27                          1   
3                        27                          1   
4                        27                          1   
5                        27                          1   
6                        27                          1   
7                        27                          1   
8                        27                          1   
9                        27                          1   

   stays_in_weekend_nights  stays_in_week_nights  adults  ...  deposit_type  \
0                        0                     0       2  ...    No Deposit   
1                        0                     0       2  ...    No Deposit   
2                        0                     1       1  ...    No Deposit   
3                        0                     1       1  ...    No Deposit   
4                        0                     2       2  ...    No Deposit   
5                        0                     2       2  ...    No Deposit   
6                        0                     2       2  ...    No Deposit   
7                        0                     2       2  ...    No Deposit   
8                        0                     3       2  ...    No Deposit   
9                        0                     3       2  ...    No Deposit   

   agent company days_in_waiting_list customer_type    adr  \
0    NaN     NaN                    0     Transient    0.0   
1    NaN     NaN                    0     Transient    0.0   
2    NaN     NaN                    0     Transient   75.0   
3  304.0     NaN                    0     Transient   75.0   
4  240.0     NaN                    0     Transient   98.0   
5  240.0     NaN                    0     Transient   98.0   
6    NaN     NaN                    0     Transient  107.0   
7  303.0     NaN                    0     Transient  103.0   
8  240.0     NaN                    0     Transient   82.0   
9   15.0     NaN                    0     Transient  105.5   

   required_car_parking_spaces  total_of_special_requests  reservation_status  \
0                            0                          0           Check-Out   
1                            0                          0           Check-Out   
2                            0                          0           Check-Out   
3                            0                          0           Check-Out   
4                            0                          1           Check-Out   
5                            0                          1           Check-Out   
6                            0                          0           Check-Out   
7                            0                          1           Check-Out   
8                            0                          1            Canceled   
9                            0                          0            Canceled   

  reservation_status_date  
0     

In [11]:
df['hotel'].count()

119390

In [12]:
df['hotel'].value_counts()

City Hotel      79330
Resort Hotel    40060
Name: hotel, dtype: int64

out of 119390 hotel counts, 79330 are city hotel and 40060 are Resort hotel


In [52]:
for i in df.columns:
    print(i,"----VALUE COUNTS----")
    print(df[i].value_counts())
    print('\n\n')
    
  

hotel ----VALUE COUNTS----
City Hotel      79330
Resort Hotel    40060
Name: hotel, dtype: int64



is_canceled ----VALUE COUNTS----
0    75166
1    44224
Name: is_canceled, dtype: int64



lead_time ----VALUE COUNTS----
0      6345
1      3460
2      2069
3      1816
4      1715
       ... 
458       1
371       1
737       1
435       1
387       1
Name: lead_time, Length: 479, dtype: int64



arrival_date_year ----VALUE COUNTS----
2016    56707
2017    40687
2015    21996
Name: arrival_date_year, dtype: int64



arrival_date_month ----VALUE COUNTS----
August       13877
July         12661
May          11791
October      11160
April        11089
June         10939
September    10508
March         9794
February      8068
November      6794
December      6780
January       5929
Name: arrival_date_month, dtype: int64



arrival_date_week_number ----VALUE COUNTS----
33    3580
30    3087
32    3045
34    3040
18    2926
21    2854
28    2853
17    2805
20    2785
29    2763
42    2756
31

In [53]:
df.replace("?",np.nan,inplace=True)

In [56]:

df.isnull().sum()

hotel                                  0
is_canceled                            0
lead_time                              0
arrival_date_year                      0
arrival_date_month                     0
arrival_date_week_number               0
arrival_date_day_of_month              0
stays_in_weekend_nights                0
stays_in_week_nights                   0
adults                                 0
children                               4
babies                                 0
meal                                   0
country                              488
market_segment                         0
distribution_channel                   0
is_repeated_guest                      0
previous_cancellations                 0
previous_bookings_not_canceled         0
reserved_room_type                     0
assigned_room_type                     0
booking_changes                        0
deposit_type                           0
agent                              16340
company         

Missing values were already replaced by NaN in the source dataset..


In [55]:
df.replace("NaN",np.nan,inplace=True)
df.replace(" ",np.nan,inplace=True)
df.replace("Null",np.nan,inplace=True)
df.replace("-",np.nan,inplace=True)
df.isnull().sum()

hotel                                  0
is_canceled                            0
lead_time                              0
arrival_date_year                      0
arrival_date_month                     0
arrival_date_week_number               0
arrival_date_day_of_month              0
stays_in_weekend_nights                0
stays_in_week_nights                   0
adults                                 0
children                               4
babies                                 0
meal                                   0
country                              488
market_segment                         0
distribution_channel                   0
is_repeated_guest                      0
previous_cancellations                 0
previous_bookings_not_canceled         0
reserved_room_type                     0
assigned_room_type                     0
booking_changes                        0
deposit_type                           0
agent                              16340
company         

There are still a few columns that have mssing values that are replaced by NaN..

In [62]:
df['children'].value_counts()

0.0     110796
1.0       4861
2.0       3652
3.0         76
10.0         1
Name: children, dtype: int64

Children Column represents how many children does a family has...
Some of the rows are missing, So I will replace them with Mean of the children column.

In [64]:
df.replace(np.nan,df['children'].mean(),inplace=True)

In [66]:
df['children'].isnull().sum()

0

Since the result is zero this means that all the NaN values are now replace with the mean.

In [13]:
df.isnull()

hotel  is_canceled  lead_time  arrival_date_year  arrival_date_month  \
0       False        False      False              False               False   
1       False        False      False              False               False   
2       False        False      False              False               False   
3       False        False      False              False               False   
4       False        False      False              False               False   
...       ...          ...        ...                ...                 ...   
119385  False        False      False              False               False   
119386  False        False      False              False               False   
119387  False        False      False              False               False   
119388  False        False      False              False               False   
119389  False        False      False              False               False   

        arrival_date_week_number  arrival_date_day_of_month  \
0                          False                      False   
1                          False                      False   
2                          False                      False   
3                          False                      False   
4                          False                      False   
...                          ...                        ...   
119385                     False                      False   
119386                     False                      False   
119387                     False                      False   
119388                     False                      False   
119389                     False                      False   

        stays_in_weekend_nights  stays_in_week_nights  adults  ...  \
0                         False                 False   False  ...   
1                         False                 False   False  ...   
2                         False                 False   False  ...   
3                         False                 False   False  ...   
4                         False                 False   False  ...   
...                         ...                   ...     ...  ...   
119385                    False                 False   False  ...   
119386                    False                 False   False  ...   
119387                    False                 False   False  ...   
119388                    False                 False   False  ...   
119389                    False                 False   False  ...   

        deposit_type  agent  company  days_in_waiting_list  customer_type  \
0              False   True     True                 False          False   
1              False   True     True                 False          False   
2              False   True     True                 False          False   
3              False  False     True                 False          False   
4              False  False     True                 False          False   
...              ...    ...      ...                   ...            ...   
119385         False  False     True                 False          False   
119386         False  False     True                 False          False   
119387         False  False     True                 False          False   
119388         False  False     True                 False          False   
119389         False  False     True                 False          False   

          adr  required_car_parking_spaces  total_of_special_requests  \
0       False                        False                      False   
1       False                        False                      False   
2       False                        False                      False   
3       False                        False                      False   
4       False                        False                      False   
...       ...                          ...                        ...   
119385  False

In [16]:
df.isnull().sum()

hotel                                  0
is_canceled                            0
lead_time                              0
arrival_date_year                      0
arrival_date_month                     0
arrival_date_week_number               0
arrival_date_day_of_month              0
stays_in_weekend_nights                0
stays_in_week_nights                   0
adults                                 0
children                               4
babies                                 0
meal                                   0
country                              488
market_segment                         0
distribution_channel                   0
is_repeated_guest                      0
previous_cancellations                 0
previous_bookings_not_canceled         0
reserved_room_type                     0
assigned_room_type                     0
booking_changes                        0
deposit_type                           0
agent                              16340
company         

In [36]:
df.dtypes.value_counts()

int64      16
object     12
float64     4
dtype: int64